### PR TITLE
add proxy ssg helpers

### LINKED ISSUE #2450: proxy API for SSG Helpers
### Describe the feature you'd like to request

Add function for using the proxy API for SSG helpers.

### Describe the solution you'd like to see

```ts
const ssg = await createProxySSGHelpers({
  router: appRouter,
  ctx: {},
  transformer: superjson,
});

ssg.post.byId.prefetch(1)

```

### Desribe alternate solutions

none

### Additional information

Related issue
- #2440

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by sachinraja**
Closes #2450

cc @juliusmarminge

## 🎯 Changes
```ts
import { createProxySSGHelpers } from '@trpc/react/ssg'
const ssg = createProxySSGHelpers({
    router: appRouter,
    ctx: {},
})

await ssg.post.list.prefetch()
```

I also switched back to `rollup-plugin-typescript2` because the `@rollup/plugin-typescript` was erroring that there was an excessively deep type when there wasn't.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/reactjs/ssg-helpers.md [hunk 1]
```diff
@@ -39,17 +32,15 @@ import superjson from 'superjson';
...
     ctx: {},
     transformer: superjson,
   });
   const id = context.params?.id as string;
 
   // Prefetch `post.byId`
-  await ssg.fetchQuery('post.byId', {
-    id,
-  });
+  await ssg.post.ById.fetch({ id });
```
Comment (KATT) [inline www/docs/reactjs/ssg-helpers.md:43]:
```suggestion
  await ssg.post.byId.fetch({ id });
```

File: packages/react/src/ssg/index.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export { createSSGHelpers } from './ssg';
```
Comment (KATT) [inline packages/react/src/ssg/index.ts:1]: Deprecate this

**Review by KATT — APPROVED** [review]
Looks awesome!

I'd do a couple of things:

- [ ] Update this code to use the new helper https://github.com/trpc/trpc/blob/d33ab65ea244efb5a34afcfb3b2ff1f81bb875e4/examples/next-prisma-todomvc/src/pages/%5Bfilter%5D.tsx#L371-L390
- [ ] Update this code to use the new helper https://github.com/trpc/trpc/blob/d33ab65ea244efb5a34afcfb3b2ff1f81bb875e4/examples/next-prisma-starter/src/pages/index.tsx#L95-L119
- [ ] Add `@deprecated` on the old functions

File: packages/react/src/ssg/index.ts (hunk 2 above)
Comment (juliusmarminge) [reply packages/react/src/ssg/index.ts:1]: The old one is still needed to use with old, "string-keyed" procedures

Comment (KATT) [reply packages/react/src/ssg/index.ts:1]: Yes, but we should have a deprecation warning when you do

Comment (KATT) [reply packages/react/src/ssg/index.ts:1]: Merging as is and moved my comment to #2450

**Pull request merged**